src/Native/Unix/System.Native/pal_process.c

```diff
@@ -187,6 +197,17 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
         goto done;
     }
 
+#ifndef HAVE_PIPE2
+    // We do not have pipe2(); take the lock to emulate it race free.
```

Comment:
It would be nice to keep the description of the problem from below. I do not think it is obvious what it is.

Reply:
Quite right; I ended up having to hammer the words pretty hard but there you go.